### PR TITLE
Remove fixes.tsv serialization from NullAway serialization service

### PR DESCRIPTION
**Pull request by nimakarimipour**
This PR removes the now-redundant serialization of fixes.tsv following updates to errors.tsv (#643). Previously, `fixes.tsv` recorded the locations of all non-null variables that participated in pseudo-assignments from nullable to non-null types, triggering NullAway errors. However, there was no direct link between these non-null locations and the corresponding errors. #643 addressed this by enhancing error serialization to include the relevant non-null location with each reported error. As a result, `fixes.tsv` is no longer needed, and [NullAwayAnnotator](https://github.com/ucr-riple/NullAwayAnnotator) now relies solely on `errors.tsv`.

This PR removes all serialization related to `fixes.tsv` and updates unit test expected output accordingly. Each unit test was adjusted to verify serialized errors, preserving their value and functionality.

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/fixserialization/adapters/SerializationV4Adapter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * <p>Updates to previous version (version 3):
+ *
+ * <ul>
+ *   <li>Fixes are not serialized anymore in a separate file. Since the current version of error
+ *       structure contains all the necessary information to recreate the fix, there is no need to
+ *       serialize the fix separately.
+ * </ul>
+ */
+public class SerializationV4Adapter extends SerializationV3Adapter {
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/fixserialization/adapters/SerializationV4Adapter.java:36]: Can you say more on why we need to bump the serialization version?  I feel like annotator will still work even with V3 serialized state so there is no compatibility issue?  No big harm here, just want to understand

**Review by msridhar — COMMENTED** [review]
Looks great!  One question below.  Also, have you tested a local build of NullAway with the Annotator to make sure this doesn't break anything?

**Comment by msridhar** [comment]
@nimakarimipour please let me know when this is ready for another review

File: nullaway/src/main/java/com/uber/nullaway/fixserialization/adapters/SerializationV4Adapter.java (hunk 1 above)
Comment (nimakarimipour) [reply nullaway/src/main/java/com/uber/nullaway/fixserialization/adapters/SerializationV4Adapter.java]: I removed version 4, as I initially thought any serialization change would require a version update. However, Annotator works perfectly fine with version 3, and actually, updating to version 4 causes it to crash with an "unsupported version" error, which would require an update to Annotator. Overall, I believe this update is unnecessary.

4a7cd5cd711c1183669c0d5d51142bed457ea61b

**Comment by nimakarimipour** [comment]
@msridhar I just had an end to end test with this version of NullAway and Annotator and could produce the exact same result with NullAway version `0.12.1`. I think this is ready for another review.

**Comment by nimakarimipour** [comment]
@msridhar One more note: I believe the package name is currently `fixserialization`, or we’re using `FixSerializationConfig`. It might make more sense to rename these to `serialization` and `SerializationConfig` for clarity. If that sounds good, I can create a separate PR to make the update.

**Review by msridhar — APPROVED** [review]
LGTM!

**Comment by msridhar** [comment]
> @msridhar One more note: I think the package name is currently `fixserialization`, or we’re using `FixSerializationConfig`. It might make more sense to rename these to `serialization` and `SerializationConfig` for clarity. If that sounds good, I can create a separate PR to make the update.

I think it's better to leave as is?  To make clear this is serializing state related to fixing of NullAway errors?

**Pull request merged**